src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/Metadata/IModelBindingMessages.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Metadata
+{
+    /// <summary>
+    /// Messages for errors the model binding system detects.
+    /// </summary>
+    public interface IModelBindingMessages
+    {
+        /// <summary>
+        /// Error message the model binding system adds when a property with an associated
+        /// <c>BindRequiredAttribute</c> is not bound.
+        /// </summary>
+        /// <value>Default <see cref="string"/> is "A value for the '{0}' property was not provided.".</value>
+        Func<object, string> MissingBindRequiredValueResource { get; }
```

Comment:
Within the MVC repo, we don't currently have many `Func<...>` properties. (Most `Func` mentions are method parameters.) We have one `Func<..., bool>` property named `PropertyFilter` and another 3 or 4 named `...Delegate` or `...DelegateAsync`. So does `MissingBindRequiredValueDelegate` (and similar for the other two) win?

If so, should the interface be named `IModelBindingMessageDelegates`? (I'm fine w/ `IModelBindingMessageProvider` but am wondering about consistency.)